src/som/interpreter/objectstorage/ObjectTransitionSafepoint.java

```diff
@@ -59,6 +60,19 @@ private static Assumption create() {
   @TruffleBoundary
   public void register() {
     phaser.register();
+    try {
```

Comment:
@daumayr I am not sure I am getting the full logic here.

We can have the following situations:
- register while no safepoint is done
- register while safepoint is starting
- register while safepoint is in progress

Can we distinguish the last two? only by looking at `transitionsInProgress`, no?